client/extensions/woocommerce/state/ui/promotions/edits-reducer.js

```diff
@@ -0,0 +1,69 @@
+/**
+ * External dependencies
+ */
+import { uniqueId, isEqual } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import { createReducer } from 'state/utils';
+import {
+	WOOCOMMERCE_PROMOTION_EDIT,
+	WOOCOMMERCE_PROMOTION_EDIT_CLEAR,
+} from 'woocommerce/state/action-types';
+import { getBucket } from '../helpers';
+
+export default createReducer( null, {
+	[ WOOCOMMERCE_PROMOTION_EDIT ]: editPromotionAction,
+	[ WOOCOMMERCE_PROMOTION_EDIT_CLEAR ]: clearPromotionEditsAction,
+} );
+
+function editPromotionAction( edits, action ) {
+	const { promotion, data } = action;
+	const prevEdits = edits || {};
+	const prevPromotion = promotion || { id: { placeholder: uniqueId( 'promotion_' ) } };
+	const bucket = getBucket( promotion );
+	const array = editPromotion( prevEdits[ bucket ], prevPromotion, data );
```

Comment:
This variable name caused me to pause a bit, do you think we should have something more descriptive? Maybe `promotionsArray`

Reply:
Sure, I renamed it and the other one.